packages/next-app/src/components/FilterDropdown.tsx

```diff
@@ -0,0 +1,88 @@
+"use client";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { ChevronDownIcon, EraserIcon } from "@radix-ui/react-icons";
+import { useState } from "react";
+
+import { toCammelCase, useAdminTools } from "#/contexts/AdminToolsContext";
+
+export interface IFilter {
+  name: string;
+  options: string[];
+}
+
+interface IFilterDropdown {
+  filters: IFilter[];
+}
+
+export function FilterDropdown({ filters }: IFilterDropdown) {
+  const { changeSelectedFilters, clearSelectedFilter } = useAdminTools();
+  return (
+    <DropdownMenu.Root>
+      <DropdownMenu.Trigger asChild>
+        <button
+          className="flex items-center gap-2 rounded-[4px] bg-white px-2 text-sm leading-none text-gray-400 outline-none"
+          aria-label="Customise options"
+        >
+          <span>Filter by</span>
+          <ChevronDownIcon />
+        </button>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Portal>
+        <DropdownMenu.Content
+          sideOffset={5}
+          className="rounded-md bg-white px-5 py-2 will-change-[opacity,transform] data-[side=top]:animate-slideDownAndFade data-[side=right]:animate-slideLeftAndFade data-[side=bottom]:animate-slideUpAndFade data-[side=left]:animate-slideRightAndFade"
+        >
+          {filters.map((filter) => {
+            const [filterValue, setFilterValue] = useState("");
+
+            function handleSelectFilter(value: string) {
+              setFilterValue(value);
+              changeSelectedFilters(toCammelCase(filter.name), value);
```

Comment:
do we need different cases for any reason? This is super error prone in general but needed especially when interfacing between different code conventions, which doesn't seem to be the case here (plz correct me if I'm wrong). If that's the case `#should` avoid needing to change this to camelCase at first and just have `filter.name` as what should be the parameter.

Reply:
the `filter.name` is for example `Operation Responsible` so with the camel case transformation I can filter all the actions with 

``` js

data.actions.filter(
          (item) =>
            item.operationResponsible === selectedFilters.operationResponsible
        )
```

Reply:
I think with the [enum suggestion ](https://github.com/bleu-studio/balancer-pool-metadata/pull/48#discussion_r1149283728) this can be solved